__tests__/app.test.js

```diff
@@ -0,0 +1,41 @@
+const request = require("supertest");
+const db = require("../db/connection.js");
+const { app } = require("../app");
+
+const seed = require("../db/seeds/seed.js");
+const testData = require("../db/data/test-data/index.js");
+
+beforeEach(() => seed(testData));
+
+afterAll(() => db.end());
+
+describe("Error 404: Non-existent route", () => {
+  test("non-existent path", () => {
+    return request(app)
+      .get("/api/top")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body).toEqual({ msg: "Path Not Found" });
+      });
+  });
+});
+
+describe("GET/api/topics", () => {
+  test("200: should return array of all the topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body: { topics } }) => {
+        expect(topics).toBeInstanceOf(Object);
```

Comment:
toBeInstanceOf is useful but obsolete in this case if you're using a forEach (which only works on Arrays) later on. Not a major issue though.

Your use of ObjectContaining is great for when we start adding categories later on, thus avoiding brittle tests. good work 👍

Reply:
Thanks for the feedback, appreciate it!